rate-providers/registry.json

```diff
@@ -578,6 +578,15 @@
       "warnings": [""],
       "factory": "0xe548a29631f9E49830bE8edc22d407b2D2915F31",
       "upgradeableComponents": []
+    },
+    "0x998DE64cB90EdF3d205CFDB864E199fDA4d55710": {
+      "asset": "0xd09ACb80C1E8f2291862c4978A008791c9167003",
+      "name": "TreeHouseRateProvider",
+      "summary": "safe",
+      "review": "./ChainLinkRateProvider.md",
+      "warnings": [],
+      "factory": "",
+      "upgradeableComponents": []
```

Comment:
Looks like the rate provider was deployed by a factory: [0x26dEc0e6a4249F28e0f16A1a79808bF9ba308310](https://arbiscan.io/address/0x26dec0e6a4249f28e0f16a1a79808bf9ba308310)

https://arbiscan.io/tx/0x12aa6638a911e085f44890ad051f54ceb756a41a1a27635356a1e68a444cff64

Also it looks like [tEth](https://arbiscan.io/address/0xd09ACb80C1E8f2291862c4978A008791c9167003#code) on arb may be upgradeable so should it have entries for `upgradeableComponents` ?

Reply:
I have added the factory. For adding upgradeable components I usually only focus on the apparent contracts involved in the call-chain. The call to `getRate` never touches the Treehouse contract so I do not put it into the upgradeable components. If you think it is imporant to be included, you can add it as part of your review.